### PR TITLE
multiline descriptions support

### LINKED ISSUE #207: Multiline option and command descriptions
What about to replace (https://github.com/visionmedia/commander.js/blob/master/index.js#L729)
```js
      return pad(option.flags, width)
        + '  ' + option.description;
```
with something like
```js
      return pad(option.flags, width)
        + '  ' + option.description.split('\n').join('\n  ' + Array(width+1).join(' '));
```
or
```js
      return pad(option.flags, width)
        + '  ' + option.description.replace(/^/gm, pad('', width+2)).replace(/^\s+/, '');
```

And here the same problem with commands:
https://github.com/visionmedia/commander.js/blob/master/index.js#L759

### PR DESCRIPTION
**Pull request by zxqfox**
fixes #207

### CODE REVIEW

**Comment by kimmobrunfeldt** [comment]
I would very much like to have this feature

**Comment by SomeKittens** [comment]
I'm in favor of this - can you merge with master and bring things up to date?

**Comment by zxqfox** [comment]
Any chance to get this merged?

**Comment by zxqfox** [comment]
Added tests also.

@zhiyelee Please take a look. There is a little change which would make our life much easier.

File: index.js [hunk 1]
```diff
@@ -943,7 +943,7 @@ Command.prototype.helpInformation = function() {
   }
   var usage = [
     ''
-    ,'  Usage: ' + cmdName + ' ' + this.usage()
+    ,'  Usage: ' + cmdName + ' ' + offset(this.usage(), 9)
```
Comment (zxqfox) [inline index.js]: Hardcode?

**Comment by zhiyelee** [comment]
@zxqfox Yep, this feature sounds good to me, and I'll have a look at this later.  A little busy recently..

**Comment by zxqfox** [comment]
@zhiyelee Nice ;-) Feel free to point me If something is wrong or PR will need another rebase.
Some requirement in sinon doesn't work with node 0.6 ;-(

**Comment by zxqfox** [comment]
Fixed tests, polished code a little. PR is ready ;-)

File: index.js [hunk 2]
```diff
@@ -1030,6 +1047,9 @@ function outputHelpIfNecessary(cmd, options) {
     if (options[i] == '--help' || options[i] == '-h') {
       cmd.outputHelp();
       process.exit(0);
+
+      // used for test flow only
+      options.splice(i, 1);
```
Comment (zxqfox) [inline index.js]: Just fyi: The only thing I don't really like here. But these `process.exit`s makes me mad.

Comment (SomeKittens) [inline index.js]: Yeah, I don't see a way around this either.

**Comment by SomeKittens** [comment]
All good!  Thanks so much.

**Comment by zxqfox** [comment]
Thanks! ;-)

**Pull request merged**